rfd/0073-public-image-registry.md

```diff
@@ -0,0 +1,249 @@
+
+## Details
+Teleport will use Amazon ECR and Amazon ECR Public in order to host public container images. The multiple stage registry pipeline will allow Teleport to test and verify images internally before promoting them to our customers. As of authoring this RFD, Amazon ECR Public lacks support for [vulnerability scanning and tag immutabilty](https://github.com/aws/containers-roadmap/issues/1288). The two-stage pipeline will allow us to leverage these features in the internal repository before pushing to the public. 
+
+**What about name squatting on other container registry platforms?**
+
+A not yet finished RFD on third-party artifact mirroring will address concerns regarding name squatting on other container registry platforms. See [Artifact Mirroring](https://github.com/gravitational/teleport/commit/2262efbb25463ccc135553d43293f6d8aee22ba2).
+
+### Scope
+
+This RFD will focus on the infrastructure, security, and observability of the replacement registry. It will also include a deprecation and migration plan for the existing Quay.io repositories.
+
+#### In Scope
+* Infrastructure plans with [example terraform](#appendix-a-example-terraform)
+* Migration and Deprecation plan
+
+#### Out of Scope
+* Image Signing w/ Cosign
+* Mirroring of images to third-party registries. See [Artifact Mirroring](https://github.com/gravitational/teleport/commit/2262efbb25463ccc135553d43293f6d8aee22ba2).
+
+### Infrastructure
+```
+             ┌─────────────────────────────────────────────────────────┐
+             │                                                         │
+             │    ┌───────────┐                   ┌──────────────────┐ │
+   Tag / Push│    │           │     Promotion     │                  │ │
+─────────────┼─►  │  AWS ECR  │  ──────────────►  │  AWS ECR Public  │ │◄───────── public.ecr.aws/gravitational/teleport
+             │    │           │                   │                  │ │
+             │    └───────────┘                   └──────────────────┘ │
+             │                                                         │
+             │AWS Account: teleport-prod                               │
+             └─────────────────────────────────────────────────────────┘
+```
+
+The infrastructure for this will live in the [cloud-terraform](https://github.com/gravitational/cloud-terraform) repository. The terraform for the `teleport-prod` account can be found [here](https://github.com/gravitational/cloud-terraform/tree/main/teleport-team/prod). Using AWS ECR and ECR Public allow us to rely on their managed infrastructure which reduces the operational complexity while enforcing our own security policies and allowing us to better audit changes to the environment. For more information on the pros and cons of alternatives, see [alternatives](#alternatives).
+
+### Security
```

Comment:
I'd like to see a discussion of which users will have access to read/write from which repos.  I recognize this rehashes Artifact Storage Standards to a degree, but I believe it is worth restating in order to have this RFD stand on its own.

Reply:
Add a short discussion on the main types of users with permissions: service accounts for CI/CD, teleport employees and release engineers in https://github.com/gravitational/teleport/pull/13165/commits/5fe26fa323ddbfa2daabfd84d93a85c077c4a0a6. 

Should I go into more specifics?

Reply:
I think https://github.com/gravitational/teleport/commit/5fe26fa323ddbfa2daabfd84d93a85c077c4a0a6 is plenty of context. Thanks!